distribution/openrct2.d.ts

```diff
@@ -510,6 +675,598 @@ declare global {
         "waterraise" |
         "watersetheight";
 
+
+    interface GameActionArgs {
+        flags?: number; // see GAME_COMMAND in openrct2/Game.h
+    }
+
+    interface BalloonPressArgs extends GameActionArgs {
+        id: number;
+    }
+
+    interface BannerPlaceArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        z: number;
+        direction: number;
+        object: number;
+        primaryColour: number;
+    }
+
+    interface BannerRemoveArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        z: number;
+        direction: number;
+    }
+
+    interface BannerSetColourArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        z: number;
+        direction: number;
+        primaryColour: number;
+    }
+
+    interface BannerSetNameArgs extends GameActionArgs {
+        id: number;
+        name: string;
+    }
+
+    interface BannerSetStyleArgs extends GameActionArgs {
+        id: number;
+        type: number; // 0: primary colour, 1: secondary colour: 2: no entry
+        parameter: number; // primary colour | secondary colour | 0: disable, 1: enable
+    }
+
+    interface ChangeMapSizeArgs extends GameActionArgs {
+        targetSizeX: number;
+        targetSizeY: number;
+    }
+
+    interface ClearSceneryArgs extends GameActionArgs {
+        itemsToClear: number; // Bit mask. 1: small scenery and walls, 2: large scenery, 4: footpaths.
+    }
+
+    interface ClimateSetArgs extends GameActionArgs {
+        climate: number; // 0: cool and wet, 1: warm, 2: hot and dry, 3: cold
+    }
+
+    interface FootpathAdditionPlaceArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        z: number;
+        object: number;
+    }
+
+    interface FootpathAdditionRemoveArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        z: number;
+    }
+
+    interface FootpathPlaceArgs extends GameActionArgs {
+        x: number;
+        y: number;
+        z: number;
+        direction: number; // direction or 0xFF
```

Comment:
This use the Direction type instead of number, right? I don't think the documentation is consistent about it, but it should be used here (and potentially other places)

Reply:
Direction is not enough, maybe `Direction | 0xFF`.
The path construction can be used as a cursor tool or by clicking the buttons left, right, up, down, ....
In the first case, the value `0xFF` is used as direction.

Reply:
Also, I personally dislike the use of number unions in cases where the numbers are still arithmetic (and not just enums). Suppose you want to turn something by 90 degrees, you would write `newDirection = (oldDirection + 1) & 3` or `newDirection = (oldDirection + 1) % 4` and afaik both do not recognise that the result is again compatible with `Direction`.

Reply:
agreed. I dont think it needs to be changed. In the cpp code we do have some helper methods but its still a bit messy.